homeassistant/components/egps/switch.py

```diff
@@ -0,0 +1,82 @@
+"""Switch implementation for EGPS Platform."""
+
+from typing import Any
+
+from pyegps import __version__ as PYEGPS_VERSION
+from pyegps.exceptions import EgpsException
+from pyegps.powerstrip import PowerStrip
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add EGPS sockets for passed config_entry in HA."""
+    powerstrip: PowerStrip = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        EGPowerStripSocket(powerstrip, socket)
+        for socket in range(powerstrip.numberOfSockets)
+    )
+
+
+class EGPowerStripSocket(SwitchEntity):
+    """Represents a socket of an Energenie-Socket-Strip."""
+
+    _attr_device_class = SwitchDeviceClass.OUTLET
+    _attr_should_poll = True
+    _attr_has_entity_name = True
+
+    def __init__(self, dev: PowerStrip, socket: int) -> None:
+        """Initiate a new socket."""
+        self._dev = dev
+        self._socket = socket
+        self._state = STATE_OFF
+
+        self._attr_name = f"Socket {socket}"
```

Comment:
It probably makes sense to make that translatable using `translation_keys` (and maybe `translation_placeholders`). Check [homewizard](https://github.com/home-assistant/core/blob/dev/homeassistant/components/homewizard/sensor.py) for an example.